### ISSUE TITLE
Google2 doesn't work

### ISSUE THREAD

**Issue by zandaq** [ciaranj/connect-auth#73]
Hi, 
I've modified a little bit Express Facebook example to work with Google2, but it doesn't work and returns no error. To be precise, the code below returns: {"scopedUsers":{}} null

```
...
            express.cookieParser(),
            express.session({ secret: 'session secret' }),
            auth([
                auth.Google2({appId : "appId", appSecret: "appsecret", scope: "email", callback: "https://domain/users"})
            ]),
...
api.get('/users', function(req, res){
    req.authenticate([req.param('method')], function(error, authenticated) {
    if(authenticated) {
      res.end("<html><h1>Hello Google user:" + JSON.stringify( req.getAuthDetails() ) + ".</h1></html>")
    }
    else {
      res.end(JSON.stringify(req.getAuthDetails())+' '+JSON.stringify(error));
    }
   });
});
api.get('/somepage', function(req, res){
  var sign_in_link= "/users?method=google2&redirectUrl=" + escape(req.url);
  if( req.isAuthenticated() ) {
    res.end('<html><body><h1>Signed in with Google</h1></body></html>')
  }
  else {
    res.end('<html><body><a href="'+ sign_in_link + '">Sign in with Google</a></body></html>')
  }
});
...
```

I've also tried modifications from here: https://github.com/ciaranj/connect-auth/pull/70 but it doesn't work either.

**Comment by ciaranj** [comment]
Hmmm have you tried the example app that ships with connect-auth. Sometimes providers change how things work so that app is a good out of the box measure for that issue :)

**Comment by zandaq** [comment]
I've tried the example app from my localhost (set up callbacks as http://localhost) and it doesn't work, is it supposed to be that way?

**Comment by zandaq** [comment]
I'm more curious as to why it doesn't return any error. It does return errors if you use wrong credentials or a user doesn't allow sending data, but not when everything seems ok.

**Comment by rahulkgupta** [comment]
I'm also having issues. I set the scope to "https://www.googleapis.com/auth/userinfo.profile".
However, authenticated is returning false. 

An extra note is that I'm running this on my local host and that I'm using client ID for installed applications

**Comment by pixeldrew** [comment]
I too am unable to get OAuth2 to work with Google, the error I get is: 

Phase 2/2 : Requesting an OAuth access token.
Error retrieving the contact details =>{"statusCode":404,"data":"Contact not found."}

Incidentally, why do I need to give address book access to the application? I'm afraid my users wouldn't be comfortable with that. Should I just be using OAuth1 for using google just as the authentication mechanism?

**Comment by pixeldrew** [comment]
Seems OAuth1 with google also has the same requirement. Is there anyway I can just use the Simple API Access key to authenticate a user using connect-auth?

**Comment by ciaranj** [comment]
@pixeldrew the reason for the address book access is so the 'strategy' can populate the 'username' property of the session.. ultimately it wouldn't be neccessary, and given the problems everyone appears to have with it (but my google account seems fine ! ) , it may be worth just not populating the user id inside the strategy, and requiring a first-login-handler to handle the username population :(

**Comment by ciaranj** [comment]
commit 2e549558ec74d3affdb65146b0c13554077536de should resolve this issue. (included in 0.4.3 and 0.5.1 above)
